### PR TITLE
use val_percent_check in validation step

### PR DESCRIPTION
**Pull request by cdump**
Looks like `val_percent_check` only used when calculating `self.total_batches` for progressbar limit, so when you set `val_percent_check=0.01` and run `trainer.fit(model)` full validation set will be used at the end of train epoch and tqdm progress bar will be corrupted (overflow):
```112it [00:22,  7.21it/s, batch_nb=45, dice=0.0788, epoch=1, gpu=0, tng_loss=0.528, v_nb=33]  ```

This patch fixes it, and only `val_percent_check` percent of validation set used in validation

### CODE REVIEW

**Comment by williamFalcon** [comment]
@cdump good catch!

**Comment by williamFalcon** [comment]
@cdump https://gitter.im/PyTorch-Lightning/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge
